tensorflow/python/ops/metrics_impl.py

```diff
@@ -929,14 +929,20 @@ def compute_mean_iou(name):
       cm_diag = math_ops.to_float(array_ops.diag_part(total_cm))
       denominator = sum_over_row + sum_over_col - cm_diag
 
+      # The mean is only computed over classes that appear in the
+      # label or prediction tensor. If the denominator is 0, we need to
+      # ignore the class.
+      num_valid_entries = math_ops.reduce_sum(math_ops.cast(
+          math_ops.not_equal(denominator, 0), dtype=dtypes.float32))
+
       # If the value of the denominator is 0, set it to 1 to avoid
       # zero division.
       denominator = array_ops.where(
           math_ops.greater(denominator, 0),
           denominator,
           array_ops.ones_like(denominator))
       iou = math_ops.div(cm_diag, denominator)
-      return math_ops.reduce_mean(iou, name=name)
+      return math_ops.reduce_sum(iou, name=name)/num_valid_entries
```

Comment:
We could divide by zero here if the function is called without any predictions (or with all predictions masked out). Should we add a check for this case and raise an exception ourselves?

Reply:
I think we should add a cond check for zero valid entries and return 0.0 in that case (which is I think what reduce_mean does)

Reply:
I added a check that would return 0. If you would like, I can also add a test to verify the correct behaviour. Let me know.